src/cow-react/modules/trade/docs/README.md

```diff
@@ -0,0 +1,38 @@
+# Trade module
+
+## Problem statement
+
+Since we have at least three trade widgets we need to unify their logic and avoid code duplication  ([DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)). To do this, we need to define what is specific and what is common.
+At the same time, strike a balance between code cohesion and solution scalability.
+
+## Shared language
+
+- `TradeWidget` - Widget to create CoW orders. Its configuration can create many different ways to create orders, for example Swap, Limit Orders and Advanced orders will be instances of it
+- `TradeRawState` - DTO derived from URL (`/1/USDT/WETH?recipient=0x000`) and persisted into localStorage ([ref](../../../modules/trade/types/TradeRawState.ts))
+- `TradeDerivedState` - a model that contains `CurrencyAmount` and `Currency` instances. This state is derived from TradeRawState ([ref](../../../modules/trade/types/TradeDerivedState.ts))
+
```

Comment:
Some small suggestions

```suggestion
- `Trade widget`: Widget to create CoW orders. Some examples would be the `SwapWidget` or `LimitOrdersWidget`. Normally widget implementations will make use of `TradeWidget`. 
- `AbstractTradeWidget`: Abstraction of a Trade Widget, it contains most of the shared parts of the widgets: main layout, sell token, the recipient and some modals. It will have in its parametrization some slots where other trade widgets will be able to inject components i.e. `LimitOrdersWidget` will be able to add the limit price.
- `AbstractTradeUrlState`: DTO derived from URL (`/1/USDT/WETH?recipient=0x000`) and persisted into localStorage ([ref](../../../modules/trade/types/TradeRawState.ts))
- `AbstractTradeState`: A model that contains `CurrencyAmount` and `Currency` instances. This state is derived from TradeRawState ([ref](../../../modules/trade/types/TradeDerivedState.ts))
```

Reply:
Great suggestions! Thank you!
It doesn't fully match with the current implementation, if you don't mind I will keep it as it is and will update the doc after code changes.

Reply:
Created an issue: https://github.com/cowprotocol/cowswap/issues/2556